plugins/chat/app/models/category_channel.rb

```diff
@@ -21,7 +21,26 @@ def title(_ = nil)
     name.presence || category.name
   end
 
-  def slug
-    title.truncate(100).parameterize
+  def ensure_slug
+    return if title.blank?
+    slug_title = self.title.strip
+
+    if self.slug.present?
+      # if we don't unescape it first we strip the % from the encoded version
+      slug = SiteSetting.slug_generation_method == "encoded" ? CGI.unescape(self.slug) : self.slug
+      self.slug = Slug.for(slug, "", method: :encoded)
+
+      if self.slug.blank?
+        errors.add(:slug, :invalid)
+      elsif SiteSetting.slug_generation_method == "ascii" && !CGI.unescape(self.slug).ascii_only?
+        errors.add(:slug, I18n.t("category_channel.errors.slug_contains_non_ascii_chars"))
+      elsif duplicate_slug?
+        errors.add(:slug, "is already in use")
```

Comment:
This is something we could surface to end user and wouldn't get translated right?

Reply:
Yes good point, I will fix this + the category one in this same PR.